admin/workspace-providers/deployment/ec2.md

```diff
@@ -0,0 +1,69 @@
+---
+title: EC2
+description: Learn how to deploy a workspace provider to an EC2 cluster.
+---
+
+This article walks you through the process of deploying a workspace provider to
+an EC2 instance.
+
+## Prerequisites
+
+You must have an
+[**AWS access key ID** and **secret access key**](https://docs.aws.amazon.com/general/latest/gr/aws-sec-cred-types.html#access-keys-and-secret-access-keys).
+
+We recommend having the [AWS CLI](https://aws.amazon.com/cli/) installed and
+configured as well.
+
+## Creating a workspace provider
+
+1. Log into Coder as a site manager, and go to **Manage** > **Workspace
+   providers**.
+
+1. In the top-right next to **Create Kubernetes Provider**, click on the **down
+   arrow** and select **Create Amazon EC2 Provider**.
+
+1. Provide a **name** to identify the provider.
+
+1. Provide the requested configuration details to connect Coder to your AWS
+   account:
+
+   - **Access key ID**: the AWS access key associated with your account
+   - **Secret access key**: the AWS secret access key associated with your
+     account
+   - **AWS region ID**: The AWS region where the EC2 instances should be created
+   - **AWS availability zone**: The AWS availability zone associated with the
+     region where the EC2 instances are created. You can find an available zone
+     by running `aws ec2 describe-availability-zones --region region-name`
+
+1. Provide the networking options:
+
+   - VPC ID: the VPC network to which instances should be attached. If you leave
+     this field empty, Coder uses the default VPC ID in the specified region for
+     your EC2 instances
+   - Subnet ID: the
+     [ID of the subnet](https://docs.aws.amazon.com/managedservices/latest/userguide/find-subnet.html)
+     associated with your VPC and availability zone
+
+1. Specify the Amazon Machine Image configuration you want to be used when
+   launching workspaces:
```

Comment:
AMI ID, AMI username, root volume size and docker volume size are required. All fields in this section have default values though, but for AMI you will have to click a button to use the defaults as I mentioned in another comment.

Volume sizing recommendations:
- Root volume should be at least the size of the AMI, with at least a couple gigs to spare for stuff like logs etc.
- Docker volume size should be at least the size of the workspace image and much more, it will store anything written to outside of the home directory as well (such as logs, docker containers in the workspace etc.)
    - Big estimate might be max image size in the org + 10-20GB?